internal/pkg/cli/pipeline_init.go

```diff
@@ -157,35 +161,46 @@ func newInitPipelineOpts(vars initPipelineVars) (*initPipelineOpts, error) {
 
 // Validate returns an error if the optional flag values passed by the user are invalid.
 func (o *initPipelineOpts) Validate() error {
-	// This command must be executed in the app's workspace because the pipeline manifest and buildspec will be created and stored.
-	if err := validateInputApp(o.wsAppName, o.appName, o.store); err != nil {
-		return err
-	}
 	return nil
 }
 
 // Ask prompts for required fields that are not passed in and validates them.
 func (o *initPipelineOpts) Ask() error {
-	if o.repoURL != "" {
-		if err := o.validateURL(o.repoURL); err != nil {
-			return err
-		}
-	} else {
-		if err := o.selectURL(); err != nil {
+	// This command must be executed in the app's workspace because the pipeline manifest and buildspec will be created and stored.
+	if err := validateInputApp(o.wsAppName, o.appName, o.store); err != nil {
+		return err
+	}
+	o.appName = o.wsAppName
+
+	if o.name == "" {
+		if err := o.askPipelineName(); err != nil {
 			return err
 		}
 	}
+	if err := validatePipelineName(o.name, o.appName); err != nil {
+		return err
+	}
 
-	if len(o.environments) > 0 {
-		if err := o.validateEnvs(); err != nil {
+	// TODO: validate not a duplicate
```

Comment:
What does this mean exactly? (This is my personal preference) but I find TODOs a bit intimidating. I can never tell if it's something that has to happen or it's just a nice to have and I'm safe to touch this code?

If it's just a nice-to-have, can we remove it please? 🥺

Reply:
good catch, I meant to remove that. It's something I need to do in my next PR :)